src/jit/flowgraph.cpp

```diff
@@ -4761,19 +4761,37 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr, IL_OFFSET codeSize, BYTE*
                 __fallthrough;
 
             case CEE_CALLI:
+
+                // CEE_CALLI should not be inlined if the call indirect target has a calling convention other than
+                // CORINFO_CALLCONV_DEFAULT. In the case where we have a no-marshal CALLI P/Invoke we end up calling
+                // the IL stub. We don't NGEN these stubs, so we'll have to JIT an IL stub for a trivial func.
+                // It's almost certainly a better choice to leave out the inline candidate so we can generate an inlined 
+                // call frame.
+
+                // Consider making this only for not force inline.
+                if (makeInlineObservations)
+                {
+                    CORINFO_SIG_INFO sig;
+                    info.compCompHnd->findSig(info.compScopeHnd, getU4LittleEndian(codeAddr),
```

Comment:
Probably should add the paranoia check here that there are actually enough bytes of IL left to read the token.